packages/integrations/src/vue.ts

```diff
@@ -1,103 +1,344 @@
+import { Integrations as APMIntegrations, Span as SpanClass } from '@sentry/apm';
 import { EventProcessor, Hub, Integration } from '@sentry/types';
-import { getGlobalObject, isPlainObject, logger } from '@sentry/utils';
+import { basename, getGlobalObject, logger, timestampWithMs } from '@sentry/utils';
 
-/** JSDoc */
-interface Metadata {
+/** Global Vue object limited to the methods/attributes we require */
+interface VueInstance {
+  config?: {
+    errorHandler?(error: Error, vm?: ViewModel, info?: string): void; // tslint:disable-line:completed-docs
+  };
+  mixin(hooks: { [key: string]: () => void }): void; // tslint:disable-line:completed-docs
+  util: {
+    warn(...input: any): void; // tslint:disable-line:completed-docs
+  };
+}
+
+/** Representation of Vue component internals */
+interface ViewModel {
   [key: string]: any;
-  componentName?: string;
-  propsData?: {
+  $root: object;
+  $options: {
     [key: string]: any;
+    name?: string;
+    propsData?: { [key: string]: any };
+    _componentTag?: string;
+    __file?: string;
+    $_sentryPerfHook?: boolean;
   };
+  $once(hook: string, cb: () => void): void; // tslint:disable-line:completed-docs
+}
+
+// tslint:enable:completed-docs
+
+/** Vue Integration configuration */
+interface IntegrationOptions {
+  /** Vue instance to be used inside the integration */
+  Vue: VueInstance;
+
+  /**
+   * When set to `false`, Sentry will suppress reporting of all props data
+   * from your Vue components for privacy concerns.
+   */
+  attachProps: boolean;
+  /**
+   * When set to `true`, original Vue's `logError` will be called as well.
+   * https://github.com/vuejs/vue/blob/c2b1cfe9ccd08835f2d99f6ce60f67b4de55187f/src/core/util/error.js#L38-L48
+   */
+  logErrors: boolean;
+
+  /** When set to `true`, enables tracking of components lifecycle performance. */
+  tracing: boolean;
+
+  /** {@link TracingOptions} */
+  tracingOptions: TracingOptions;
+}
+
+/** Vue specific configuration for Tracing Integration  */
+interface TracingOptions {
+  /**
+   * Decides whether to track components by hooking into its lifecycle methods.
+   * Can be either set to `boolean` to enable/disable tracking for all of them.
```

Comment:
Can we defer the option to "track all"? It seems like a sharp edge for me.

Plus, `TracingOptions.trackComponents: false` somehow conflicts with `IntegrationOptions.tracing`

Reply:
`trackComponents` will be effectively ignored if you don't set `tracing: true`.
Regarding tracking everything, I'm not sure why we'd want to postpone it tbh cc @HazAT